### PR TITLE
Allow us to override the shader versions for pixel and vertex shaders individually.

### PR DESCRIPTION
**Pull request by Xenthio**
My first PR to this repo, nothing seems broken in my testing!

### CODE REVIEW

File: src/d3d9/d3d9_adapter.cpp [hunk 1]
```diff
@@ -538,12 +538,19 @@ namespace dxvk {
     // Max Stream Stride
     pCaps->MaxStreamStride           = 508; // bytes
 
+
     const uint32_t majorVersion = options.shaderModel;
     const uint32_t minorVersion = options.shaderModel != 1 ? 0 : 4;
 
+    const uint32_t PSMajorVersion = options.PSShaderModel;
```
Comment (MarkEHenderson) [inline src/d3d9/d3d9_adapter.cpp:545]: To ease merging with the original dxvk, all changes to the original dxvk files need to be wrapped with comments, like:
```
// NV-DXVK start: DLFG integration
  code.goes.here();
// NV-DXVK end
```

File: src/d3d9/d3d9_adapter.cpp [hunk 2]
```diff
@@ -538,12 +538,19 @@ namespace dxvk {
     // Max Stream Stride
     pCaps->MaxStreamStride           = 508; // bytes
 
+
```
Comment (MarkEHenderson) [inline src/d3d9/d3d9_adapter.cpp:541]: extra new line, please remove

File: src/d3d9/d3d9_options.h [hunk 3]
```diff
@@ -57,8 +57,10 @@ namespace dxvk {
     /// Limit frame rate
     int32_t maxFrameRate;
 
-    /// Set the max shader model the device can support in the caps.
+    /// Set the max shader model the device can support in the caps. these default to d3d9.shaderModel's value if not specifically set  
     int32_t shaderModel;
+    int32_t PSShaderModel;
```
Comment (MarkEHenderson) [inline src/d3d9/d3d9_options.h:62]: please match the case of the other variables.  They appear to be using lowerCamelCase, so these variables should be `psShaderModel`  and `vsShaderModel`

File: src/d3d9/d3d9_options.h [hunk 4]
```diff
@@ -57,8 +57,10 @@ namespace dxvk {
     /// Limit frame rate
     int32_t maxFrameRate;
 
-    /// Set the max shader model the device can support in the caps.
+    /// Set the max shader model the device can support in the caps. these default to d3d9.shaderModel's value if not specifically set
```
Comment (MarkEHenderson) [inline src/d3d9/d3d9_options.h:60]: I would move the new comment to be right above psShaderModel instead of above shaderModel

**Review by MarkEHenderson — COMMENTED** [review]
Please remember to add yourself to the credits! see `dxvk_imgui_about.cpp`

**Comment by Xenthio** [comment]
Alright! hopefully thats all good :D

**Comment by MarkEHenderson** [comment]
Please squash your commits into a single commit (makes merging this a lot easier).  

There are a bunch of guides online for how to do that, this one seems decent:
https://www.git-tower.com/learn/git/faq/git-squash (the interactive rebase part)

**Comment by Xenthio** [comment]
> Please squash your commits into a single commit (makes merging this a lot easier).
> 
> There are a bunch of guides online for how to do that, this one seems decent: https://www.git-tower.com/learn/git/faq/git-squash (the interactive rebase part)

Alright done!

**Comment by MarkEHenderson** [comment]
Just merged this to our internal repo. It should show up in Github within the next hour or so.

**Pull request closed without merging**